src/gardenlinux/github/__main__.py

```diff
@@ -0,0 +1,836 @@
+                        for key, value in data["details"].items():
+                            output += f"    {key}: {value}\n"
+                    elif platform == "gcp" and "details" in data:
+                        # For GCP, move details up to same level as flavor
+                        for key, value in data["details"].items():
+                            output += f"  {key}: {value}\n"
+                    elif "gallery_images" in data or "marketplace_images" in data:
+                        if data.get("gallery_images"):
+                            output += "  gallery_images:\n"
+                            for img in data["gallery_images"]:
+                                output += f"    - hyper_v_generation: {img['hyper_v_generation']}\n"
+                                output += f"      azure_cloud: {img['azure_cloud']}\n"
+                                output += f"      image_id: {img['image_id']}\n"
+                        if data.get("marketplace_images"):
+                            output += "  marketplace_images:\n"
+                            for img in data["marketplace_images"]:
+                                output += f"    - hyper_v_generation: {img['hyper_v_generation']}\n"
+                                output += f"      urn: {img['urn']}\n"
+
+                output += "```\n\n"
+                output += "</details>\n\n"
+
+            output += "</details>\n\n"
+
+        output += "</details>\n\n"
+
+    return output
+
+
+def download_metadata_file(
+    s3_artifacts, cname, version, commitish_short, artifacts_dir
+):
+    """
+    Download metadata file (s3_metadata.yaml)
+    """
+    LOGGER.debug(
+        f"{s3_artifacts=} | {cname=} | {version=} | {commitish_short=} | {artifacts_dir=}"
+    )
+    release_object = list(
+        s3_artifacts._bucket.objects.filter(
+            Prefix=f"meta/singles/{cname}-{version}-{commitish_short}"
+        )
+    )[0]
+    LOGGER.debug(f"{release_object.bucket_name=} | {release_object.key=}")
+
+    s3_artifacts._bucket.download_file(
+        release_object.key, artifacts_dir.joinpath(f"{cname}.s3_metadata.yaml")
+    )
+
+
+def download_all_metadata_files(version, commitish):
+    repo = Repo(".")
+    commit = repo.commit(commitish)
+    flavors_data = commit.tree["flavors.yaml"].data_stream.read().decode("utf-8")
+    flavors = FlavorsParser(flavors_data).filter(only_publish=True)
+
+    local_dest_path = Path("s3_downloads")
+    if local_dest_path.exists():
+        shutil.rmtree(local_dest_path)
+    local_dest_path.mkdir(mode=0o755, exist_ok=False)
+
+    s3_artifacts = S3Artifacts(GARDENLINUX_GITHUB_RELEASE_BUCKET_NAME)
+
+    commitish_short = commitish[:8]
+
+    for flavor in flavors:
+        cname = CName(flavor[1], flavor[0], "{0}-{1}".format(version, commitish_short))
+        LOGGER.debug(f"{flavor=} {version=} {commitish=}")
+        # Filter by image variants - only download if the flavor matches one of the variants
```

Comment:
This means we have a hardcoded list of flavors? Why parse the `flavors.yaml` in this case at all if we already know what we want to see. Hardcoded logic is not really future proof anyway. Does this really make sense at all?

Please consider replacing hardcoded logic with e.g. calling `FlavorsParser.filter(include_only_patterns=image_variants_with_wildcards_added)` to get the same functionality in a at least somewhat future proof way.